hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSLeafQueue.java

```diff
@@ -557,28 +599,33 @@ private boolean preemptContainerPreCheck() {
         getFairShare());
   }
 
-  /**
-   * Is a queue being starved for its min share.
-   */
-  @VisibleForTesting
-  boolean isStarvedForMinShare() {
-    return isStarved(getMinShare());
+  private Resource minShareStarvation() {
```

Comment:
This seems convoluted.  why not:

boolean starved = false;
Resource starvation = none();
if (Resources.greaterThan(policy.getResourceCalculator(),
    scheduler.getClusterResource(), getDemand(), getMinShare())) {
  starvation = Resources.subtract(getMinShare(), getResourceUsage());
  starved = Resources.greaterThan(policy.getResourceCalculator(),
    scheduler.getClusterResource(), starvation, none());
}

Nets out to the same thing, but the logic isn't as hard (for me) to follow.